exchanges/lbank/lbank.go

```diff
@@ -0,0 +1,510 @@
+}
+
+// USD2RMBRate finds USD-CNY Rate
+func (l *Lbank) USD2RMBRate() (ExchangeRateResponse, error) {
+	var resp ExchangeRateResponse
+	path := fmt.Sprintf("%s/v%s/%s", lbankAPIURL, lbankAPIVersion, lbankUSD2CNYRate)
+	return resp, l.SendHTTPRequest(path, &resp)
+}
+
+// GetWithdrawConfig gets information about withdrawals
+func (l *Lbank) GetWithdrawConfig(assetCode string) ([]WithdrawConfigResponse, error) {
+	var resp []WithdrawConfigResponse
+	params := url.Values{}
+	params.Set("assetCode", assetCode)
+	path := fmt.Sprintf("%s/v%s/%s?%s", lbankAPIURL, lbankAPIVersion, lbankWithdrawConfig, params.Encode())
+	return resp, l.SendHTTPRequest(path, &resp)
+}
+
+// Withdraw sends a withdrawal request
+func (l *Lbank) Withdraw(account, assetCode, amount, memo, mark string) (WithdrawResponse, error) {
+	var resp WithdrawResponse
+	params := url.Values{}
+	params.Set("account", account)
+	params.Set("assetCode", assetCode)
+	params.Set("amount", amount)
+	if memo != "" {
+		params.Set("memo", memo)
+	}
+	if mark != "" {
+		params.Set("mark", mark)
+	}
+	path := fmt.Sprintf("%s/v%s/%s", lbankAPIURL, lbankAPIVersion, lbankWithdraw)
+	return resp, l.SendAuthHTTPRequest("POST", path, params, &resp)
+}
+
+// RevokeWithdraw cancels the withdrawal given the withdrawalID
+func (l *Lbank) RevokeWithdraw(withdrawID string) (RevokeWithdrawResponse, error) {
+	var resp RevokeWithdrawResponse
+	params := url.Values{}
+	if withdrawID != "" {
+		params.Set("withdrawId", withdrawID)
+	}
+	path := fmt.Sprintf("%s/v%s/%s?", lbankAPIURL, lbankAPIVersion, lbankRevokeWithdraw)
+	return resp, l.SendAuthHTTPRequest("POST", path, params, &resp)
+}
+
+// GetWithdrawalRecords gets withdrawal records
+func (l *Lbank) GetWithdrawalRecords(assetCode, status, pageNo, pageSize string) (WithdrawalResponse, error) {
+	var resp WithdrawalResponse
+	params := url.Values{}
+	params.Set("assetCode", assetCode)
+	params.Set("status", status)
+	params.Set("pageNo", pageNo)
+	params.Set("pageSize", pageSize)
+	path := fmt.Sprintf("%s/v%s/%s", l.APIUrl, lbankAPIVersion, lbankWithdrawalRecords)
+	return resp, l.SendAuthHTTPRequest("POST", path, params, &resp)
+}
+
+// ErrorCapture captures errors
+func ErrorCapture(intermediary json.RawMessage) error {
+	var capErr ErrCapture
+	err := json.Unmarshal(intermediary, &capErr)
+	if err == nil && capErr.Error != 0 {
+		msg, ok := errorCodes[capErr.Error]
+		if !ok {
+			return fmt.Errorf("undefined code please check api docs for error code definition: %v", capErr.Error)
+		}
+		return errors.New(msg)
+	}
+	return nil
+}
+
+// SendHTTPRequest sends an unauthenticated HTTP request
+func (l *Lbank) SendHTTPRequest(path string, result interface{}) error {
+	var intermediary json.RawMessage
+	err := l.SendPayload(http.MethodGet, path, nil, nil, &intermediary, false, false, l.Verbose, l.HTTPDebugging)
+	if err != nil {
+		return err
+	}
+	err = ErrorCapture(intermediary)
+	if err != nil {
+		return err
+	}
+	return json.Unmarshal(intermediary, result)
```

Comment:
By this point (and the same with SendAuthHTTPRequest) we have unmarshaled the data 3 times 

It might be better to move ErrCapture as an anonymous struct with omitempty (that way if its not there it wont try unmarshal and the values will default to 0/false)

```go

type TickerResponse struct {
	ErrCapture  `json:",omitempty"`
	Symbol    string `json:"symbol"`
	Timestamp int64  `json:"timestamp"`
	Ticker    Ticker `json:"ticker"`
}

func ErrorCapture(code int) error {
	msg, ok := errorCodes[code]
	if !ok {
		return fmt.Errorf("undefined code please check api docs for error code definition: %v", code)
	}
	return errors.New(msg)
}

func (l *Lbank) GetTicker(symbol string) (TickerResponse, error) {
	var t TickerResponse
	params := url.Values{}
	params.Set("symbol", symbol)
	path := fmt.Sprintf("%s/v%s/%s?%s", l.APIUrl, lbankAPIVersion, lbankTicker, params.Encode())

	err := l.SendHTTPRequest(path, &t)
	if err != nil {
		return t, err
	}

	if t.Error != 0 {
		return t, ErrorCapture(t.Error)
	}

	return t, nil
}

func (l *Lbank) SendHTTPRequest(path string, result interface{}) error {
	return l.SendPayload(http.MethodGet, path, nil, nil, &result, false, false, l.Verbose, l.HTTPDebugging)
}

```

it does add additional information to the struct (int/bool) but at the trade off of not having to do expensive unmarshaling multiple times